### ISSUE TITLE
coverage report with skip covered: column widths still based on hidden filenames

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#524]
*Originally reported by* **Anthony Sottile (Bitbucket: [asottile](https://bitbucket.org/asottile), GitHub: [asottile](https://github.com/asottile))**

----------------------------------------

Ends up with a little silly things like:

```
py35 runtests: commands[2] | coverage report -m --show-missing --fail-under 100
Name                                                                         Stmts   Miss Branch BrPart  Cover   Missing
------------------------------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------------
TOTAL                                                                        13639      0   1163      0   100%

432 files skipped due to complete coverage.
```

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/524

**Comment by nedbat** [comment]
*Original comment by* **Anthony Sottile (Bitbucket: [asottile](https://bitbucket.org/asottile), GitHub: [asottile](https://github.com/asottile))**

----------------------------------------

Here's the start of a patch, I'd need to write tests though (and relearn mercurial)

```diff
$ git diff
diff --git a/coverage/summary.py b/coverage/summary.py
index [b0fa71a (bb)](https://bitbucket.org/ned/coveragepy/commits/b0fa71a)..cdd8851 [100644 (bb)](https://bitbucket.org/ned/coveragepy/commits/100644)
--- a/coverage/summary.py
+++ b/coverage/summary.py
@@ -28,7 +28,16 @@ class SummaryReporter(Reporter):
         file_reporters = self.find_file_reporters(morfs)
 
         # Prepare the formatting strings, header, and column sorting.
-        max_name = max([len(fr.relative_filename()) for fr in file_reporters] + [5])
+        max_name = max(
+            [5] + [
+                len(fr.relative_filename())
+                for fr in file_reporters
+                if not (
+                    self.config.skip_covered and
+                    self.coverage._analyze(fr).numbers.n_missing == 0
+                )
+            ]
+        )
         fmt_name = u"%%- %ds  " % max_name
         fmt_err = u"%s   %s: %s"
         fmt_skip_covered = u"\n%s file%s skipped due to complete coverage."
```

```
$ coverage report --include="a_really*,test.py"
Name                                                                                              Stmts   Miss  Cover
---------------------------------------------------------------------------------------------------------------------
a_really_really_really_really_really_really_really_really_really_really_really_long_filename.py       2      0   100%
test.py                                                                                               4      1    75%
---------------------------------------------------------------------------------------------------------------------
TOTAL                                                                                                 6      1    83%

$ coverage report --include="a_really*,test.py" --skip-cov
Name      Stmts   Miss  Cover
-----------------------------
test.py       4      1    75%
-----------------------------
TOTAL         6      1    83%

1 file skipped due to complete coverage.
```

**Comment by nedbat** [comment]
*Original comment by* **Loic Dachary (Bitbucket: [dachary](https://bitbucket.org/dachary), GitHub: [dachary](https://github.com/dachary))**

----------------------------------------

I'll write the tests for your fine patch, unless you already did ?

**Comment by nedbat** [comment]
*Original comment by* **Anthony Sottile (Bitbucket: [asottile](https://bitbucket.org/asottile), GitHub: [asottile](https://github.com/asottile))**

----------------------------------------

Ah lol I totally forgot about this patch.  Yeah I have yet to write tests for it -- looking at my working directory I had just gotten to the point where I was frustrated with mercurial and gave up.

**Comment by nedbat** [comment]
*Original comment by* **Loic Dachary (Bitbucket: [dachary](https://bitbucket.org/dachary), GitHub: [dachary](https://github.com/dachary))**

----------------------------------------

@asottile here it is : https://bitbucket.org/ned/coveragepy/pull-requests/98/report-must-ignore-skipped-files-524/diff . Would you have time to review it ?

**Comment by nedbat** [comment]
Fixed in [6d6ddf50dd7f (bb)](https://bitbucket.org/ned/coveragepy/commits/6d6ddf50dd7f), thanks Loïc!

**Comment by nedbat** [comment]
This fix was released as part of Coverage.py 4.3.
